code/modules/industrial_lift/tram/tram_remote.dm

```diff
@@ -0,0 +1,154 @@
+			direction = TRAMCTRL_INBOUND
+	update_appearance()
+	balloon_alert(user, "[direction ? "< inbound" : "outbound >"]")
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+///set safety bypass
+/obj/item/tram_remote/CtrlClick(mob/user)
+	switch(mode)
+		if(TRAMCTRL_SAFE)
+			mode = TRAMCTRL_FAST
+		if(TRAMCTRL_FAST)
+			mode = TRAMCTRL_SAFE
+	update_appearance()
+	balloon_alert(user, "mode: [mode ? "fast" : "safe"]")
+
+/obj/item/tram_remote/examine(mob/user)
+	. = ..()
+	if(!tram_ref)
+		. += "There is an X showing on the display."
+		. += "Left-click a tram request button to link."
+		return
+	. += "The arrow on the display is pointing [direction ? "inbound" : "outbound"]."
+	. += "The rapid mode light is [mode ? "on" : "off"]."
+	if (!COOLDOWN_FINISHED(src, tram_remote))
+		. += "The number on the display shows [DisplayTimeText(COOLDOWN_TIMELEFT(src, tram_remote), 1)]."
+	else
+		. += "The display indicates ready."
+	. += "Left-click to dispatch tram."
+	. += "Right-click to toggle direction."
+	. += "Ctrl-click to toggle safety bypass."
+
+/obj/item/tram_remote/update_icon_state()
+	. = ..()
+	if(!tram_ref)
+		icon_state = "tramremote_nis"
+		return
+	switch(direction)
+		if(TRAMCTRL_INBOUND)
+			icon_state = "tramremote_ib"
+		if(TRAMCTRL_OUTBOUND)
+			icon_state = "tramremote_ob"
+
+/obj/item/tram_remote/update_overlays()
+	. = ..()
+	if(mode == TRAMCTRL_FAST)
+		. += mutable_appearance(icon, "tramremote_emag")
+
+/obj/item/tram_remote/attack_self(mob/user)
+	if (!COOLDOWN_FINISHED(src, tram_remote))
+		balloon_alert(user, "cooldown: [DisplayTimeText(COOLDOWN_TIMELEFT(src, tram_remote), 1)]")
+		return FALSE
+	if(try_force_tram(user))
+		COOLDOWN_START(src, tram_remote, 2 MINUTES)
+
+///send our selected commands to the tram
+/obj/item/tram_remote/proc/try_force_tram(mob/user)
+	var/datum/lift_master/tram/tram_part = tram_ref?.resolve()
+	if(!tram_part)
+		balloon_alert(user, "no tram linked!")
+		return FALSE
+	if(tram_part.controls_locked || tram_part.travelling) // someone else started already
+		balloon_alert(user, "tram busy!")
+		return FALSE
+	var/tram_id = tram_part.specific_lift_id
+	var/destination_platform = null
+	var/platform = 0
+	switch(direction)
+		if(TRAMCTRL_INBOUND)
+			platform = clamp(tram_part.idle_platform.platform_code - 1, 1, INFINITY)
+		if(TRAMCTRL_OUTBOUND)
+			platform = clamp(tram_part.idle_platform.platform_code + 1, 1, INFINITY)
+	if(platform == tram_part.idle_platform.platform_code)
+		balloon_alert(user, "invalid command!")
+		return FALSE
+	for (var/obj/effect/landmark/tram/destination as anything in GLOB.tram_landmarks[tram_id])
+		if(destination.platform_code == platform)
+			destination_platform = destination
+			break
+	if(!destination_platform)
+		balloon_alert(user, "invalid command!")
+		return FALSE
+	else
+		switch(mode)
+			if(TRAMCTRL_FAST)
+				tram_part.tram_travel(destination_platform, TRAMCTRL_FAST)
+			if(TRAMCTRL_SAFE)
+				tram_part.tram_travel(destination_platform, TRAMCTRL_SAFE)
```

Comment:
The second parameter here is `rapid`, so it's not obvious the values of these macros are important since you check them as bools. This should be `rapid = TRUE` and `rapid = FALSE` instead to make that obvious.